--- package/libs/libbsd/Makefile	2026-01-15 00:45:21.630742846 +0800
+++ package/libs/libbsd/Makefile	2026-01-15 00:45:14.698744471 +0800
@@ -30,7 +30,8 @@
  This library provides useful functions commonly found on BSD systems, and lacking on others like GNU systems, thus making it easier to port projects with strong BSD origins, without needing to embed the same code over and over again on each project.
 endef
 
-TARGET_CFLAGS += -D_LARGEFILE64_SOURCE
+TARGET_CFLAGS += -D_LARGEFILE64_SOURCE -fno-lto
+TARGET_LDFLAGS += -fno-lto
 
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/lib
