--- a/include/target.mk
+++ b/include/target.mk
@@ -270,9 +270,9 @@
     CPU_CFLAGS_ultrasparc = -mcpu=ultrasparc
   endif
   ifeq ($(ARCH),aarch64)
-    CPU_TYPE ?= generic
-    CPU_CFLAGS_generic = -mcpu=generic
-    CPU_CFLAGS_cortex-a53 = -mcpu=cortex-a53
+    CPU_TYPE ?= cortex-a53
+    CPU_CFLAGS_generic = -march=armv8-a+crc+simd+crypto -mtune=cortex-a53 -mcpu=cortex-a53+crypto
+    CPU_CFLAGS_cortex-a53 = -march=armv8-a+crc+simd+crypto -mtune=cortex-a53 -mcpu=cortex-a53+crypto
   endif
   ifeq ($(ARCH),arc)
     CPU_TYPE ?= arc700
