diff --git a/include/image.mk b/include/image.mk
index 048e750..767dec0 100644
--- a/include/image.mk
+++ b/include/image.mk
@@ -96,6 +96,12 @@ ifeq ($(CONFIG_SQUASHFS_XZ),y)
   endif
   SQUASHFSCOMP := xz $(LZMA_XZ_OPTIONS) $(BCJ_FILTER)
 endif
+ifeq ($(CONFIG_SQUASHFS_ZSTD),y)
+  SQUASHFSCOMP := zstd -Xcompression-level 16
+endif
+ifeq ($(CONFIG_SQUASHFS_LZ4),y)
+  SQUASHFSCOMP := lz4 -Xhc
+endif
 
 JFFS2_BLOCKSIZE ?= 64k 128k
 
diff --git a/tools/Makefile b/tools/Makefile
index 752a561..c4911b6 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -134,7 +134,7 @@ $(curdir)/pkgconf/compile := $(curdir)/meson/compile
 $(curdir)/quilt/compile := $(curdir)/autoconf/compile $(curdir)/findutils/compile
 $(curdir)/sdcc/compile := $(curdir)/bison/compile
 $(curdir)/squashfs3-lzma/compile := $(curdir)/lzma-old/compile
-$(curdir)/squashfs4/compile := $(curdir)/xz/compile $(curdir)/zlib/compile
+$(curdir)/squashfs4/compile := $(curdir)/xz/compile $(curdir)/zlib/compile $(curdir)/zstd/compile $(curdir)/lz4/compile
 $(curdir)/util-linux/compile := $(curdir)/bison/compile $(curdir)/automake/compile
 $(curdir)/yafut/compile := $(curdir)/cmake/compile
 
diff --git a/tools/squashfs4/Makefile b/tools/squashfs4/Makefile
index 94679bd..ac90e79 100644
--- a/tools/squashfs4/Makefile
+++ b/tools/squashfs4/Makefile
@@ -24,9 +24,9 @@ include $(INCLUDE_DIR)/host-build.mk
 define Host/Compile
 	+$(HOST_MAKE_VARS) \
 	$(MAKE) $(HOST_JOBS) -C $(HOST_BUILD_DIR)/squashfs-tools \
-		LZ4_SUPPORT=0 \
+		LZ4_SUPPORT=1 \
 		LZO_SUPPORT=0 \
-		ZSTD_SUPPORT=0 \
+		ZSTD_SUPPORT=1 \
 		GZIP_SUPPORT=1 \
 		XZ_SUPPORT=1 \
 		LZMA_XZ_SUPPORT=1 \
