diff --git a/package/mtk/drivers/warp/src/wed.c b/package/mtk/drivers/warp/src/wed.c
index 0b834388f5..d5c5fd4c28 100644
--- a/package/mtk/drivers/warp/src/wed.c
+++ b/package/mtk/drivers/warp/src/wed.c
@@ -908,7 +908,7 @@ wed_proc_handle(struct wed_entry *wed, char choice, char *arg)
 
 	case WED_PROC_TX_DYNAMIC_FREE: {
 		if(IS_WED_HW_CAP(wed, WED_HW_CAP_DYN_TXBM))
-			buf_free_task((unsigned long)wed);
+			buf_free_task(wed);
 		else
 			warp_dbg(WARP_DBG_ERR, "DYBM is not enabled!\n");
 	}

diff --git a/package/mtk/drivers/warp/src/mcu/warp_ccif.c b/package/mtk/drivers/warp/src/mcu/warp_ccif.c
index 497c7950b1..acbd3d2a1d 100644
--- a/package/mtk/drivers/warp/src/mcu/warp_ccif.c
+++ b/package/mtk/drivers/warp/src/mcu/warp_ccif.c
@@ -150,7 +150,7 @@ static void ccif_exit(struct woif_bus *bus)
 		disable_irq(ccif->irq);
 		free_irq(ccif->irq, bus);
 		if (ccif->base_addr)
-			iounmap(ccif->base_addr);
+			iounmap((void *)ccif->base_addr);
 		warp_os_free_mem(ccif);
 	}
 
